test/quantization/fx/test_model_report_fx.py

```diff
@@ -431,3 +435,301 @@ def forward(self, x):
+            for index, batch in enumerate(split_up_data):
+
+                num_tracked_so_far = getattr(model, "obs1").num_batches_tracked
+                self.assertEqual(num_tracked_so_far, index)
+
+                # get general info about the batch and the model to use later
+                batch_min, batch_max = torch.aminmax(batch)
+                current_average_range = getattr(
+                    model, "obs1"
+                ).average_batch_activation_range
+                current_epoch_min = getattr(model, "obs1").epoch_activation_min
+                current_epoch_max = getattr(model, "obs1").epoch_activation_max
+
+                # print(current_average_range,current_epoch_max,current_epoch_min)
+
+                # run input through
+                model(ex_input)
+
+                # check min and max of epoch is correctly recorded
+                self.assertEqual(
+                    getattr(model, "obs1").epoch_activation_min,
+                    min(current_epoch_min, batch_min),
+                )
+                self.assertEqual(
+                    getattr(model, "obs1").epoch_activation_max,
+                    max(current_epoch_max, batch_max),
+                )
+
+                # check that batches run correctly updated
+                self.assertEqual(
+                    getattr(model, "obs1").num_batches_tracked, num_tracked_so_far + 1
+                )
+
+                # check that average batch activation range updated correctly
+                correct_updated_value = (
+                    current_average_range * num_tracked_so_far + (batch_max - batch_min)
+                ) / (num_tracked_so_far + 1)
+                self.assertEqual(
+                    getattr(model, "obs1").average_batch_activation_range,
+                    correct_updated_value,
+                )
+
+    """Case includes:
+        all zero tensor
+        dim size = 2
+        run for 1 epoch
+        run for 10 batch
+        tests input data observer
+    """
+
+    def test_zero_tensor_errors(self):
+        # initialize the model
+        model = self.NestedModifiedSingleLayerLinear()
+
+        # generate the desired input
+        ex_input = torch.zeros((10, 1, 5))
+
+        # run it through the model and do general tests
+        self.run_model_and_common_checks(model, ex_input, 1, 1)
+
+        # make sure final values are all 0
+        self.assertEqual(getattr(model, "obs1").epoch_activation_min, 0)
+        self.assertEqual(getattr(model, "obs1").epoch_activation_max, 0)
+        self.assertEqual(getattr(model, "obs1").average_batch_activation_range, 0)
+
+        # we should get an error if we try to calculate the ratio
+        properly_errored = False
+        try:
+            ratio_val = getattr(model, "obs1").get_batch_to_epoch_ratio()
+        except ValueError:
+            properly_errored = True
+
+        self.assertEqual(properly_errored, True)
```

Comment:
we can use this: https://github.com/pytorch/pytorch/blob/master/test/test_autograd.py#L356